### PR TITLE
Include Legacy shell.nix in Flake Templates

### PR DESCRIPTION
**Pull request by kclejeune**

### CODE REVIEW

**Comment by kclejeune** [comment]
@i077 could you add this on #5?

File: python-mach-nix/shell.nix [hunk 1]
```diff
@@ -0,0 +1,7 @@
+# legacy shell.nix support flake devShell: https://nixos.wiki/wiki/Flakes
+(import (let lock = builtins.fromJSON (builtins.readFile ./flake.lock);
+in fetchTarball {
+  url =
+    "https://github.com/edolstra/flake-compat/archive/${lock.nodes.flake-compat.locked.rev}.tar.gz";
+  sha256 = lock.nodes.flake-compat.locked.narHash;
```
Comment (i077) [inline python-mach-nix/shell.nix]: The `sha256` (or any hash argument) to `fetchTarball` is optional, so we don't need to have it here or as a flake input.
This file (and the others) should really just be

```nix
# Flake's devShell for non-flake-enabled nix instances
(import (fetchTarball "https://github.com/edolstra/flake-compat/archive/master.tar.gz") {
  src = ./.;
}).shellNix.default
```
and `flake-compat` inputs should be removed from flake.nix in all the templates.

**Review by i077 — CHANGES_REQUESTED** [review]

File: python-mach-nix/shell.nix (hunk 1 above)
Comment (kclejeune) [reply python-mach-nix/shell.nix]: done

File: python-poetry/flake.nix [hunk 2]
```diff
@@ -1,30 +1,30 @@
...
+    poetry2nix = {
+      url = "github:nix-community/poetry2nix";
+      inputs.nixpkgs.follows = "nixpkgs";
+    };
+    flake-utils.url = "github:numtide/flake-utils";
   };
-  inputs.flake-utils.url = "github:numtide/flake-utils";
 
-  outputs = { self, nixpkgs, flake-utils, poetry2nix }:
+  outputs = { self, nixpkgs, flake-utils, poetry2nix, ... }:
```
Comment (i077) [inline python-poetry/flake.nix]:
```suggestion
  outputs = { self, nixpkgs, flake-utils, poetry2nix }:
```

**Review by i077 — APPROVED** [review]
Just remove that ellipsis before you merge lol

**Pull request merged**